ncloud/resource_ncloud_sourcecommit_repository.go

```diff
@@ -0,0 +1,268 @@
+
+	if err != nil {
+		logErrorResponse("resourceNcloudSourceCommitRepositoryRead", err, *id)
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to search repository",
+			Detail:   fmt.Sprintf("Unable to search repository - detail repository id : %s", *id),
+		})
+		return diags
+	}
+
+	logResponse("resourceNcloudSourceCommitRepositoryRead", repository)
+
+	if repository == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(strconv.Itoa(int(*repository.Id)))
+	d.Set("name", repository.Name)
+	d.Set("description", repository.Description)
+	d.Set("creator", repository.Created.User)
+	d.Set("git_https_url", repository.Git.Https)
+	d.Set("git_ssh_url", repository.Git.Ssh)
+	d.Set("filesafer", repository.Linked.FileSafer)
+
+	return nil
+}
+
+func resourceNcloudSourceCommitRepositoryUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*ProviderConfig)
+
+	reqParams := &sourcecommit.ChangeRepository{
+		Description: StringPtrOrNil(d.GetOk("description")),
+	}
+
+	if fileSafer, ok := d.GetOk("filesafer"); ok {
+		reqParams.Linked = &sourcecommit.CreateRepositoryLinked{
+			FileSafer: BoolPtrOrNil(fileSafer, ok),
+		}
+	}
+
+	id := ncloud.String(d.Id())
+
+	logCommonRequest("resourceNcloudSourceCommitRepositoryUpdate", reqParams)
+	_, err := config.Client.sourcecommit.V1Api.ChangeRepository(ctx, reqParams, id)
+
+	if err != nil {
+		logErrorResponse("resourceNcloudSourceCommitRepositoryUpdate", err, *id)
+		return diag.FromErr(err)
+	}
+
+	logResponse("resourceNcloudSourceCommitRepositoryUpdate", id)
+	return resourceNcloudSourceCommitRepositoryRead(ctx, d, meta)
+}
+
+func resourceNcloudSourceCommitRepositoryDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*ProviderConfig)
+
+	id := ncloud.String(d.Id())
+
+	logCommonRequest("resourceNcloudSourceCommitRepositoryDelete", *id)
+
+	if _, err := config.Client.sourcecommit.V1Api.DeleteRepository(ctx, id); err != nil {
+		logErrorResponse("resourceNcloudSourceCommitRepositoryDelete", err, *id)
+		return diag.FromErr(err)
+	}
+
+	logResponse("resourceNcloudSourceCommitRepositoryDelete", id)
+	d.SetId("")
+	return nil
+}
+
+func waitForSourceCommitRepositoryActive(ctx context.Context, d *schema.ResourceData, config *ProviderConfig, name string) error {
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"PENDING"},
+		Target:  []string{"RESOLVE"},
+		Refresh: func() (result interface{}, state string, err error) {
+			repository, err := getRepository(ctx, config, name)
+			if err != nil {
+				return nil, "", fmt.Errorf("Repository response error , name : (%s) to become activating: %s", name, err)
+			}
+			if repository == nil {
+				return name, "NULL", nil
+			}
+
+			if ncloud.StringValue(repository.Name) == name {
+				d.SetId(strconv.Itoa(*repository.Id))
+				return repository, "RESOLVE", nil
+			}
+
+			return nil, "PENDING", err
```

Comment:
```suggestion
			return nil, "PENDING", nil
```

위에서 error check 했기 때문에 여기서는 nil을 리턴해주면 될것 같습니다.
해당 라인은 상태 확인을 하고, `PENDING` 상태를 응답해서 retry하려는 목적인데 error가 nil이 아니게되면 retry하지 않고 error를 반환하게 됩니다.

Reply:
넵 nil 을 return 하도록 수정하였습니다